utils/config.go

```diff
@@ -387,6 +391,20 @@ func ValidateLdapFilter(cfg *model.Config) *model.AppError {
 	return nil
 }
 
+func ValidateLocales(cfg *model.Config) *model.AppError {
+	if len(*cfg.LocalizationSettings.AvailableLocales) > 0 {
+		for _, word := range strings.Split(*cfg.LocalizationSettings.AvailableLocales, ",") {
+			if word == *cfg.LocalizationSettings.DefaultClientLocale {
+				return nil
+			}
+		}
+
+		return model.NewLocAppError("ValidateLocales", "utils.config.validate_locale.app_error", nil, "")
+	}
+
+	return nil
```

Comment:
No, this is checking that if your `AvailableLocales` is set then it has to include whatever you have in the `DefaultClientLocale` or else this will prevent you from starting the server.

1. Your `AvailableLocales` is empty server starts no problem as you decided to allow every locale there is
2. Your `AvailableLocales` has `en` and the `DefaultClientLocale` is `en` server starts
3. `AvailableLocales` does not include `zh_TW` and your `DefaultClientLocale` is `zh_TW` server won't start